code/game/objects/items/food/pastries.dm

```diff
@@ -56,7 +56,8 @@
 
 ///Override for checkliked in edible component, because all cops LOVE donuts
 /obj/item/food/donut/proc/check_liked(fraction, mob/living/carbon/human/H)
-	if(HAS_TRAIT(H.mind, TRAIT_LAW_ENFORCEMENT_METABOLISM) && !HAS_TRAIT(H, TRAIT_AGEUSIA))
+	var/obj/item/organ/liver/liver = H.getorganslot(ORGAN_SLOT_LIVER)
+	if(!HAS_TRAIT(H, TRAIT_AGEUSIA) && liver && HAS_TRAIT(liver, TRAIT_LAW_ENFORCEMENT_METABOLISM))
```

Comment:
its a bit weird that changing your liver affects your taste

Reply:
So then this part could be put in the tongue or stomach?

Reply:
For another PR, could make a new "mind_trait" that just makes you like donuts, changing this. Or I could do this in the same one if that's not too complex.